test/unit/uk/gov/hmrc/exports/services/NotificationServiceSpec.scala

```diff
@@ -175,12 +182,59 @@ class NotificationServiceSpec extends WordSpec with MockitoSugar with ScalaFutur
         verify(submissionRepositoryMock, times(1)).updateMrn(meq(actionId), meq(mrn))
       }
 
-      trait SaveHappyPathTest extends Test {
-        when(notificationRepositoryMock.save(any())).thenReturn(Future.successful(true))
-        when(submissionRepositoryMock.updateMrn(any(), any())).thenReturn(Future.successful(Some(submission)))
+      "call NotificationRepository but not SubmissionRepository when saving an unparsed notification" in new SaveHappyPathTest {
+        notificationService.save(notificationUnparsable).futureValue
+
+        verify(notificationRepositoryMock, times(1)).save(meq(notificationUnparsable))
+        verifyNoInteractions(submissionRepositoryMock)
       }
     }
 
+    "NotificationService on reattemptParsingUnparsedNotifications" should {
+      "do nothing if no unparsed notifications exist" in new SaveHappyPathTest {
+        when(notificationRepositoryMock.findUnparsedNotifications())
+          .thenReturn(Future.successful(Seq.empty))
+
+        notificationService.reattemptParsingUnparsedNotifications().futureValue
+      }
```

Comment:
If I see correctly, this test does not contain any assertion.

Reply:
Lol, well it testing doing nothing! Fixed :)